src/playlists.ts

```diff
@@ -0,0 +1,37 @@
+import * as vscode from 'vscode';
+
+
+
+export class PlaylistsProvider implements vscode.TreeDataProvider<Playlist> {
+  constructor(private spotify: any) {}
+
+
+  getChildren(element?: Playlist): Thenable<Playlist[]> {
+    if (element) {
+      return Promise.resolve([]);
+    }
+
+    const user = this.spotify.getMe();
+
+    return Promise.resolve(this.spotify.getUserPlaylists(user.id)
+    .then((data) => {
+      return Promise.resolve(data.body.items.map((playlist: any) => new Playlist(playlist.name)));
+    },function(err) {
+      console.log('Something went wrong!', err);
```

Comment:
**Neutral:** I don't know if console messages are easily viewable if you're running the extension normally without debugging, and it might be nicer to see at least an indication that an error happened somewhere that's guaranteed to be visible to the user.
```suggestion
      vscode.window.showErrorMessage(err);
```